include/ddc/kernels/splines/spline_builder.hpp

```diff
@@ -890,4 +931,118 @@ operator()(
                 });
     }
 }
+
+template <
+        class ExecSpace,
+        class MemorySpace,
+        class BSplines,
+        class InterpolationDDim,
+        ddc::BoundCond BcLower,
+        ddc::BoundCond BcUpper,
+        SplineSolver Solver,
+        class... IDimX>
+template <class OutMemorySpace>
+std::tuple<
+        ddc::Chunk<
+                double,
+                ddc::DiscreteDomain<
+                        ddc::Deriv<typename InterpolationDDim::continuous_dimension_type>>,
+                ddc::KokkosAllocator<double, OutMemorySpace>>,
+        ddc::Chunk<
+                double,
+                ddc::DiscreteDomain<InterpolationDDim>,
+                ddc::KokkosAllocator<double, OutMemorySpace>>,
+        ddc::Chunk<
+                double,
+                ddc::DiscreteDomain<
+                        ddc::Deriv<typename InterpolationDDim::continuous_dimension_type>>,
+                ddc::KokkosAllocator<double, OutMemorySpace>>>
+SplineBuilder<
+        ExecSpace,
+        MemorySpace,
+        BSplines,
+        InterpolationDDim,
+        BcLower,
+        BcUpper,
+        Solver,
+        IDimX...>::quadrature_coefficients() const
+{
+    // Compute integrals of bsplines
+    ddc::Chunk integral_bsplines(spline_domain(), ddc::KokkosAllocator<double, MemorySpace>());
+    ddc::discrete_space<bsplines_type>().integrals(integral_bsplines.span_view());
+
+    // Remove last cell in the periodic case
+    ddc::ChunkSpan integral_bsplines_without_periodic_point
+            = integral_bsplines[ddc::DiscreteDomain<bsplines_type>(
+                    ddc::DiscreteElement<bsplines_type>(0),
+                    ddc::DiscreteVector<bsplines_type>(matrix->size()))];
```

Comment:
Can you explain a bit ? It is not clear to me reading the code that it does actually remove periodic point. `spline_domain()` is basically `full_domain()` of size `ncells() + degree()`. On the contrary `matrix->size()` equals `nbasis()` that is `ncells() + !is_periodic() * degree()`. I don't see the link with your comment "Remove last cell in the periodic case"

Reply:
Actually it is a mistake to speak about "periodic cell". The explanation is the documentation of `full_domain()` function.